bpf/bpf_lxc.c

```diff
@@ -108,6 +108,14 @@ static __always_inline int __per_packet_lb_svc_xlate_4(void *ctx, struct iphdr *
 		ret = lb4_local(get_ct_map4(&tuple), ctx, ipv4_is_fragment(ip4),
 				ETH_HLEN, l4_off, &key, &tuple, svc, &ct_state_new,
 				has_l4_header, false, &cluster_id, ext_err);
+
+#ifdef SERVICE_NO_BACKEND_RESPONSE
+		if (ret == DROP_NO_SERVICE) {
+			ep_tail_call(ctx, CILIUM_CALL_IPV4_NO_SERVICE);
+			return DROP_MISSED_TAIL_CALL;
+		}
+#endif
+
```

Comment:
With ep-routes enabled, the ICMP packet should now pass through `to-container` on the way back into the pod. Would we thus require an ingress network policy change to allow this traffic?

This feels very similar to the topic of service loopback replies ...

Reply:
A CT entry should be created for the outgoing connection to the service, so when doing policy checking the ICMP reply should be flagged as return traffic an thus not subject to any ingress policy.

Reply:
Unfortunately [lb4_local()](https://github.com/cilium/cilium/blob/4400169ba429babe013419b363d8c00e6872ff0f/bpf/lib/lb.h#L1579) currently doesn't create the RELATED entry (note the `NULL` for `map_related`). So I don't think there's any CT entry in place that would allow such ICMP traffic to pass through network policy enforcement.

Reply:
I see that depends on `ENABLE_PER_PACKET_LB` being enabled or not

Reply:
> Right, but isn't the CT entry created [here](https://github.com/cilium/cilium/blob/4400169ba429babe013419b363d8c00e6872ff0f/bpf/bpf_lxc.c#L959) before we get to the LB stage?

Nope, that part is only reached *after* selecting the backend (this CT entry tracks the client -> backend connection).

Reply:
Copied to https://github.com/cilium/cilium/issues/33041.